lib/hashie/extensions/coercion/dry_types.rb

```diff
@@ -0,0 +1,37 @@
+require 'dry-types'
+
+module Hashie
+  module Extensions
+    module Coercion
+      module DryTypes
+        module Types
+          include Dry::Types.module
+        end
+
+        def self.extended(base)
+          base.const_set('Types', Types)
+        end
+
+        def build_coercion(type)
+          if type.respond_to? :call
+            lambda do |value|
+              return type.call(value)
+            end
+          elsif type.respond_to? :coerce
+            lambda do |value|
+              return value if value.is_a? type
+              type.coerce(value)
+            end
+          elsif type.respond_to? :new
+            lambda do |value|
```

Comment:
See my comments about the other "lamdbas that just return the original". I think we can just call new.